### PR TITLE
Feat: Completed Auth and product detials page

### PR DESCRIPTION
**Pull request by vinc3nati**
- Completed Auth feature with guest login
- Added product details page

image
image
image

### CODE REVIEW

**Review by anuj-kumary — APPROVED** [review]

File: src/components/carousel/carousel.css [hunk 1]
```diff
@@ -21,26 +23,47 @@
...
 }
 
-.inner {
+/* .inner {
   white-space: nowrap;
+  width: 100% !important;
+  height: 100% !important;
   transition: transform 0.3s;
-}
+} */
```
Comment (OmKakatkar) [inline src/components/carousel/carousel.css]: Remove commented code

File: src/components/product/ProductCard.jsx [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
+  postCart,
+  postWishlist,
+} from "../../utilities/API_REQUESTS";
+import { useData } from "../../contexts/data-context";
+import { ACTIONS } from "../../utilities/constant";
 
 export const ProductCard = ({ product }) => {
   const {
+    _id,
+    id,
```
Comment (OmKakatkar) [inline src/components/product/ProductCard.jsx]: Are these 2 ids required?

**Review by OmKakatkar — APPROVED** [review]
Try to keep a PR relevant to a single feature.
Folder Structure Looks good.

**Review by shilpe26 — APPROVED** [review]
▪️The overall code looks easy to read
▪️The file structure is easy to understand
▪️Good to merge

File: src/contexts/auth-context.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      value={{ user, handleLogin, handleLogOut, handleSignUp }}
+    >
+      {children}
+    </AuthContext.Provider>
+  );
+};
+
+const useAuth = () => {
+  const context = useContext(AuthContext);
+  if (!context) throw new Error("AUth Context was not created");
```
Comment (Shah-Arsalan) [inline src/contexts/auth-context.js:71]: Oh seeing throw keyword for the first time . Something new !

**Review by Shah-Arsalan — APPROVED** [review]
Looks good to me . You can go ahead with the merge

**Review by lomesshh — APPROVED** [review]
clean code
Looks good to merge

File: src/components/carousel/carousel.css (hunk 1 above)
Comment (vinc3nati) [reply src/components/carousel/carousel.css]: sure

File: src/components/product/ProductCard.jsx (hunk 2 above)
Comment (vinc3nati) [reply src/components/product/ProductCard.jsx]: Nope, removing it

**Pull request merged**